demo/flutter/lib/main.dart

```diff
@@ -0,0 +1,249 @@
+    } finally {
+      this.setState(() {
+        isButtonDisabled = false;
+      });
+    }
+  }
+
+  void wakeWordCallback(int keywordIndex) {
+    if (keywordIndex == 0) {
+      this.setState(() {
+        rhinoText = "Wake word detected!\nListening for intent...";
+      });
+    }
+  }
+
+  void inferenceCallback(Map<String, dynamic> inference) {
+    this.setState(() {
+      rhinoText = prettyPrintInference(inference);
+    });
+
+    Future.delayed(const Duration(milliseconds: 2500), () {
+      if (isProcessing) {
+        this.setState(() {
+          rhinoText = "Listening for wake word...";
+        });
+      } else {
+        this.setState(() {
+          rhinoText = "";
+        });
+      }
+    });
+  }
+
+  void errorCallback(PvError error) {
+    print(error.message);
+  }
+
+  String prettyPrintInference(Map<String, dynamic> inference) {
+    String printText =
+        "{\n    \"isUnderstood\" : \"${inference['isUnderstood']}\",\n";
+    if (inference['isUnderstood']) {
+      printText += "    \"intent\" : \"${inference['intent']}\",\n";
+      if (inference['slots'].length > 0) {
+        printText += '    "slots" : {\n';
+        Map<String, String> slots = inference['slots'];
+        for (String key in slots.keys) {
+          printText += "        \"$key\" : \"${slots[key]}\",\n";
+        }
+        printText += '    }\n';
+      }
+    }
+    printText += '}';
+    return printText;
+  }
+
+  Future<String> _extractAsset(String resourcePath) async {
+    // extraction destination
+    String resourceDirectory = (await getApplicationDocumentsDirectory()).path;
+    String outputPath = '$resourceDirectory/$resourcePath';
+    File outputFile = new File(outputPath);
+
+    ByteData data = await rootBundle.load(resourcePath);
+    final buffer = data.buffer;
+
+    await outputFile.create(recursive: true);
+    await outputFile.writeAsBytes(
+        buffer.asUint8List(data.offsetInBytes, data.lengthInBytes));
+    return outputPath;
+  }
+
+  Future<void> _startProcessing() async {
+    if (isProcessing) {
+      return;
+    }
+
+    this.setState(() {
+      isButtonDisabled = true;
+    });
+
+    try {
+      await _picovoiceManager.start();
+      this.setState(() {
+        isProcessing = true;
+        rhinoText = "Listening for wake word...";
+      });
+    } on PvAudioException catch (ex) {
+      print("Failed to start audio capture: ${ex.message}");
+    } finally {
+      this.setState(() {
+        isButtonDisabled = false;
+      });
+    }
+  }
+
+  Future<void> _stopProcessing() async {
+    if (!isProcessing) {
+      return;
+    }
+
+    this.setState(() {
+      isButtonDisabled = true;
+    });
+
+    try {
+      await _picovoiceManager.stop();
+      this.setState(() {
+        isProcessing = false;
+        rhinoText = "";
+      });
+    } on PvAudioException catch (ex) {
+      print("Failed to start audio capture: ${ex.message}");
```

Comment:
ailed to start audio -> ailed to stop[ audio 

can this even fail?

Reply:
yep!